ethexe/service/src/tests.rs

```diff
@@ -1008,13 +1009,137 @@ async fn multiple_validators() {
     assert_eq!(res.payload, res.message_id.encode().as_slice());
 }
 
+#[tokio::test(flavor = "multi_thread")]
+#[ntest::timeout(120_000)]
+async fn tx_pool_gossip() {
+    gear_utils::init_default_logger();
+
+    let test_env_config = TestEnvConfig {
+        validators: ValidatorsConfig::Generated(2),
+        ..Default::default()
+    };
+
+    // Setup env of 2 nodes, one of them knows about the other one.
+    let mut env = TestEnv::new(test_env_config).await.unwrap();
+
+    log::info!("📗 Starting node 0");
+    let mut node0 = env.new_node(
+        NodeConfig::default()
+            .validator(env.validators[0], env.validator_session_public_keys[0])
+            .service_rpc(9505)
+            .network(None, None),
+    );
+    node0.start_service().await;
+
+    log::info!("📗 Starting node 1");
+    let mut node1 = env.new_node(
+        NodeConfig::default()
+            .validator(env.validators[1], env.validator_session_public_keys[1])
+            .network(None, node0.multiaddr.clone()),
+    );
+    node1.start_service().await;
+
+    log::info!("Populate node-0 and node-1 with 2 valid blocks");
+
+    env.observer
+        .provider()
+        .evm_mine(None)
+        .await
+        .expect("failed mining a new block");
+    env.observer
+        .provider()
+        .evm_mine(None)
+        .await
+        .expect("failed mining a new block");
+
+    // Give some time for nodes to process the blocks
+    tokio::time::sleep(Duration::from_secs(2)).await;
+    let reference_block = node0
+        .db
+        .latest_valid_block()
+        .expect("at least genesis block is latest valid")
+        .0;
+
+    // Prepare tx data
+    let signed_ethexe_tx = {
+        let sender_pub_key = env.signer.generate_key().expect("failed generating key");
+
+        let ethexe_tx = OffchainTransaction {
+            raw: RawOffchainTransaction::SendMessage {
+                program_id: H160::random(),
+                payload: vec![],
+            },
+            // referring to the latest valid block hash
+            reference_block,
+        };
+        let signature = env
+            .signer
+            .sign(sender_pub_key, ethexe_tx.encode().as_ref())
+            .expect("failed signing tx");
+        SignedOffchainTransaction {
+            signature: signature.encode(),
+            transaction: ethexe_tx,
+        }
+    };
+
+    // Send request
```

Comment:
some rpc wrapper for sending requests would be much more user-friendly

Reply:
done